### ISSUE TITLE
Need to generate training data in the json format 

### ISSUE THREAD

**Issue by ferc255** [ssu-451/project#43]
Recently I tested NN with big number of neurons (#42). It worked quite well, but the test case contained only three letters.

I think we can consider it as a starting point. There is no deal breaker to test NN with real data.

But! NN will contain 16*16 input neurons, about 150 hidden neurons and something more output neurons. It's quite many! And we can't test all the time when we run "tox" because it'd take us hours to wait for test's ending.

So I suppose we're ready for smooth transition from local working to interaction with the server. We'll teach NN on the server.

Of course, there are a lot of work to do it. And I should make a little more preparations of NN's methods (but that's another subject).

**TL;DR:**
Anyway first important thing is training dataset (TD). Look at small prototype of TD: [nn_aye.json](https://github.com/ssu-451/project/blob/master/ocr/tests/nn_aye.json)

That's a small TD for only three letters. I made it manually. Now we need the same TD with all letters.

I suggest you use only Latin letters. Don't give a fuck about Russian letters. There's not a point. It'll just slow down NN's work.

So ___the task___: you should generate json file in the [following format](https://github.com/ssu-451/project/blob/master/ocr/tests/nn_aye.json) for the following symbols:
```
ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789 .,"?'!-
```

**Assignees**: @Andy-ch, @KolesnikovaTamara, @WalkingMike 
**Labels**: feature request
**Milestone**: 2018.03.20

**Comment by Andy-ch** [comment]
If you store your loads of coefficients in a single JSON file, it will cause significant time to read or write the huge file, and it will also take a lot if RAM to store (do you really have to keep all your coefficients in RAM at all times? The server does not have a lot of RAM you know (but my desktop computer does))
And probably not only coefficients will make a huge file, but also he training dataset (btw we have the training dataset as a bunch of image files, why not retreive those pixels from them on the run?)
Anyway, we will have to do something about the huge JSON file to make it possible to read/write only a certain part of it. IMO, the perfect solution for that will be using a database, a pocket one, like Sqlite. Sqlitr databases are stored in single files, and Sqlite doesn't even have a special server process, so you can imagine how simple it is. It will require some SQL knowledge though. What do you think?

**Comment by ferc255** [comment]
This json file will be used only one time. When we start training NN, we load this json and update "train_dataset" variable. That's all! Then we don't need it at all until we want to train NN on this dataset again. 

60 letters * 16*16 * 4 bytes ~= 60 MB of RAM. Can your server deal with it?

**Comment by Andy-ch** [comment]
I thought each letter takes at least 8 bytes, so 120MB? Well, anyway, that's fine
But what about the nn's coefficients? How much will it take?

**Comment by Andy-ch** [comment]
If you need that dataset file so badly, alright then :)

**Comment by ferc255** [comment]
NN's weights is a graph that actually is a 3d array with summarily (16*16*150 + 150*60) * 8 bytes = 370 KB elements.

**Comment by ferc255** [comment]
There won't be any problems with memory.
Anyway at first, I thought about database. But then, when I realized that we need to store only 3d array, I understood that we've no need for database here.
